scripts/build/Platform/Linux/build_config.json

```diff
@@ -383,7 +383,7 @@
     "PARAMETERS": {
       "CONFIGURATION": "profile",
       "OUTPUT_DIRECTORY": "build/linux",
-      "CMAKE_OPTIONS": "-G 'Ninja Multi-Config' -DLY_PARALLEL_LINK_JOBS=4 -DLY_DISABLE_TEST_MODULES=TRUE -DLY_STRIP_DEBUG_SYMBOLS=TRUE",
+      "CMAKE_OPTIONS": "-G 'Ninja Multi-Config' -DLY_PARALLEL_LINK_JOBS=4 -DLY_DISABLE_TEST_MODULES=TRUE -DLY_STRIP_DEBUG_SYMBOLS=TRUE -DO3DE_INSTALLER_BUILD=TRUE",
```

Comment:
Instead of needing to specify the `O3DE_INSTALLER_BUILD` option when building an install layout, instead, the CMake script can query the [sdk_engine](https://github.com/o3de/o3de/blob/development/cmake/install/engine.json.in#L15) `engine.json` field using the [o3de_read_json_key](https://github.com/o3de/o3de/blob/development/cmake/O3DEJson.cmake#L51-L59) function

Reply:
I might be able to move it out of a compile rule and make this a runtime logic gate instead